ocaml/xapi/xapi_pif.ml

```diff
@@ -50,6 +50,18 @@ let refresh_internal ~__context ~self =
 				end)
 			(Opt.of_exception (fun () -> get_value ())) in
 
+	let pif_capabilities = Net.Interface.get_capabilities dbg device in
+        let get_capabilities = Db.PIF.get_capabilities ~__context ~self in
+	let capabilities = 
+		match pif_capabilities with
+			| [] -> if (List.mem "fcoe" get_capabilities) then
+					List.filter (fun x -> x <> "fcoe") get_capabilities else get_capabilities 
+			| [x] -> if not (List.mem "fcoe" get_capabilities) then
+					List.append get_capabilities [x] else get_capabilities
+			| _ -> get_capabilities
+	in
+	Db.PIF.set_capabilities ~__context ~self ~value:capabilities;
+	
```

Comment:
I'm not sure I understand this bit. I was expecting something like the following, at the end of the function:
```
maybe_update_database "capabilities"
		(Db.PIF.get_capabilities)
		(Db.PIF.set_capabilities)
		(fun () -> Net.Interface.get_capabilities dbg ~name:device)
		(String.concat "; ")
```
There is no need to do anything "fcoe" specific.